markdown_mermaidjs/markdown_mermaidjs.py

```diff
@@ -10,57 +10,99 @@
     from markdown import Markdown
 
 
-MERMAID_CODEBLOCK_START = re.compile(r"^(?P<code_block_sign>[\~\`]{3})[Mm]ermaid\s*$")
-MERMAID_JS_SCRIPT = """
+class MermaidPreprocessor(Preprocessor):
+    MERMAID_CODEBLOCK_START = re.compile(
+        r"^(?P<code_block_sign>[\~\`]{3})[Mm]ermaid\s*$"
+    )
+
+    def __init__(self, md: Markdown, icon_packs: dict = {}) -> None:
```

Comment:
```suggestion
    def __init__(self, md: Markdown, icon_packs: dict | None = None) -> None:
```

it's suggested not to use mutable in default value

Reply:
Good feedback, I've updated it (and related code)